src/main/cpp/alias/Alias.cpp

```diff
@@ -0,0 +1,77 @@
+/*
+ * Alias.cpp
+ *
+ *  Created on: Mar 11, 2017
+ *      Author: cedric
+ */
+
+#include "Alias.h"
+#include "AliasUtil.h"
+
+#include <exception>
+#include <assert.h>
+#include <math.h>
+
+namespace stride {
+namespace alias {
+
+Alias& Alias::CreateDistribution(std::vector<double> probabilities, util::Random& rng)
```

Comment:
[There ya go.](https://gist.github.com/jonathanvdc/faa38c7a8320c125d04d9d4adb756ce2) The trick's to rely on a move constructor.